### ISSUE TITLE
incorrect type - TryStatement's finalizer property can't be null

### ISSUE THREAD

**Issue by RebeccaStevens** [typescript-eslint/typescript-eslint#2912]
> Please don't ignore this template.

There doesn't seem to be a template for "@typescript-eslint/types".

**Checks**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Problem**

The `finalizer` property on the `TryStatement` node type can be `null` but the type says it can't be.

**Expected Type**

```ts
interface TryStatement extends BaseNode {
  // ...
  finalizer: BlockStatement | null;
}
```

**Actual Type**

```ts
interface TryStatement extends BaseNode {
  // ...
  finalizer: BlockStatement;
}
```

**Versions**

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/type`              | `4.11.1` |
| `TypeScript`                           | `4.1.3` |
| `node`                                 | `14.15.3` |

**Comment by bradzacher** [comment]
Happy to accept a PR!
